srcjs/shinyapp.js

```diff
@@ -473,11 +499,23 @@ var ShinyApp = function() {
   });
 
 
+  addCustomMessageHandler('busy', function(message) {
+    if (message === true) {
+      $(document).trigger('shiny:busy');
+    } else if (message === false) {
+      $(document).trigger('shiny:idle');
```

Comment:
I believe the existing lines where `shiny-busy` class is added/removed from the document element, should be moved to here.